### PR TITLE
Loosen flask version requirement to >= 2

### LINKED ISSUE #478: minimum flask version differs between website info and github code
According to your website, minimum flask version is 2.0.0.
However, by checking your code, I can see your tags >= 2.0 request flask 2.2.0 instead.
I have a large project using flask 2.1.2, and it would demand a large effort to upgrade it to flask 2.2.0.
Do you really need flask >= 2.2.0? Could it be 2.1.2?
Thanks in advance.

### PR DESCRIPTION
**Pull request by wlach**
Later versions in the Flask 2.x series changed the API a fair bit.
Loosen the version requirement to >= 2, which allows people to continue
using 2.0.x and 2.1.x. Add tests using 2.0.3 to be sure this continues
working. Fixes #478.

### CODE REVIEW

**Comment by wlach** [comment]
There's a fair number of unit test failures because the examples depend on more recent features in flask-sqlalchemy (get_or_404, paginate). I could fix them, but I'd like to wait for guidance from @greyli before continuing.

**Comment by greyli** [comment]
It's unnecessary to add a separate test env for Flask 2.0.

**Comment by wlach** [comment]
@greyli Ok I think this is ready for review! Appreciate your support and encouragement on this

File: requirements/flask20.in [hunk 1]
```diff
@@ -0,0 +1,5 @@
+authlib==1.2.1
```
Comment (greyli) [inline requirements/flask20.in:1]: The purpose of the flask20 test is to test the minimum versions of the install requirements, so this file should include the following definition:

```
# minimum install requirements
flask==2.0.0
flask-marshmallow==0.12.0
webargs==8.3.0
flask-httpauth==4.0.0
apispec==6.0.0

# for example applications
flask-sqlalchemy==2.5.1
sqlalchemy==1.4.49
marshmallow-dataclass==8.5.0
authlib==1.2.1
```

And the name `min` (for test env, requirement file, etc.) would be more appropriate.

**Review by greyli — CHANGES_REQUESTED** [review]

File: tests/test_openapi_tags.py [hunk 2]
```diff
@@ -86,6 +87,8 @@ def test_auto_tag_from_blueprint(app, client):
     assert {'name': 'Foo'} in rv.json['tags']
 
 
+@pytest.mark.skipif(flask.__version__ < '2.0.1',
+                    reason='Depends on new behaviour introduced in Flask 2.0.1')
```
Comment (wlach) [inline tests/test_openapi_tags.py:91]: Needed to add these skips as well if we're going to test with actual flask 2.0.0

**Comment by wlach** [comment]
@greyli ready for another look!

**Review by greyli — APPROVED** [review]
LGTM

**Pull request merged**